### PR TITLE
Supporting JavaScript dates for expireTime

### PR DESCRIPTION
**Pull request by alexsaves**
Made some enhancements that I hope will make things a little easier for others. Specifically:

Now supporting JavaScript dates for expireTime
Added range and type checking for unix dates
Added `getSignedRTMPUrl` for conveniently generating RTMP style urls - this was confusing for me. Hope this helps!

### CODE REVIEW

**Comment by alexsaves** [comment]
Old style string or number UNIX dates work now - as do js dates. Added some tests for all this and hinting passes.

File: README.md [hunk 1]
```diff
@@ -26,8 +26,12 @@ var params = {
   keypairId: process.env.PUBLIC_KEY,
   privateKeyString: process.env.PRIVATE_KEY,
   privateKeyPath: '/path/to/private/key',      // Optional. Use as an alternative to privateKeyString.
-  expireTime: '<epoch time when you wish the link to expire>'
+  expireTime: new Date(new Date().getTime() + (1000 * 100)) // Now + 100 seconds
```
Comment (jasonsims) [inline README.md]: I still don't really get the value add of the js Date object. You're instantiating a new Date object to get the time in ms, then using the result to instantiate another Date object, only to convert back to ms again later.

How is 
```js
{expireTime: new Date(new Date().getTime() + (1000 * 100))}
```
more straightforward than
```js
{expireTime: new Date().getTime() + (1000 * 100)}
```
Also, passing the time as ms in the first place eliminates a lot of unnecessary calls to `.getTime` throughout the rest of the code.

File: lib/CannedPolicy.js [hunk 2]
```diff
@@ -15,19 +14,56 @@ function CannedPolicy(url, expireTime) {
...
+    }
+    return JSON.stringify(policy);
+};
+
+/**
+ * Check for common mistakes with types
+ * @private
+ */
+CannedPolicy.prototype._validate = function () {
+    if (typeof this.expireTime === 'undefined') {
```
Comment (jasonsims) [inline lib/CannedPolicy.js]: The addition of a policy validation is awesome. Can you just update all of your changes here to match the 2 space indent style of the rest of the code?

File: README.md (hunk 1 above)
Comment (alexsaves) [inline README.md]: In df26f4e7af130c8dd7de73f78d513d8a55fb1105, I have simplified the readme to use a simple date constructor. I realize it's totally a matter of opinion - but I think the confusion for us was more to do with the conversion to unix time and how that should be done - which is why I added support for js Date. If you'd rather revert all that I certainly understand.

File: lib/CannedPolicy.js (hunk 2 above)
Comment (alexsaves) [inline lib/CannedPolicy.js]: Yes. Good catch. I fixed this throughout to standardize on 2-space indent (df26f4e7af130c8dd7de73f78d513d8a55fb1105)

**Comment by jasonsims** [comment]
Thanks again for the contribution. I apologize for the delay on this

**Pull request merged**